galoy-client/src/error.rs

```diff
@@ -11,3 +17,30 @@ pub enum GaloyClientError {
     #[error("GaloyClientError - Authentication: {0}")]
     Authentication(String),
 }
+
+#[derive(Debug, Deserialize)]
+pub struct InnerError {
```

Comment:
Why are these individual structs and not part of `GaloyClientError` enum?
I'm thinking the `GraphQLApi(String),` variant should simply be extended to have more fields or something like that.
Or perhaps another variant `GraphQLResponseError` but not individual structs - seems very strange.